src/main/java/org/elasticsearch/search/aggregations/bucket/significant/heuristics/ScriptHeuristic.java

```diff
@@ -62,32 +79,29 @@ public String getName() {
         }
     };
 
-    public ScriptHeuristic(ExecutableScript searchScript, String scriptLang, String scriptString, ScriptService.ScriptType scriptType, Map<String, Object> params) {
+    public ScriptHeuristic(ExecutableScript searchScript, Script script) {
```

Comment:
wondering whether `CompiledScript` should hold the `Script` and `ExecutableScript` should hold the `CompiledScript`

I would at least renamed it from `searchScript` to `executableScript`...